src/components/course/data/utils.jsx

```diff
@@ -407,10 +409,344 @@ export function shortenString(str, maxLength, suffix, separator = ' ') {
+    subscription => catalogsWithCourse.includes(subscription?.enterpriseCatalogUuid),
+  ) || [];
+
+  const hasSubscriptionsApplicableToCourse = subscriptionsApplicableToCourse.length > 0;
+  if (!hasSubscriptionsApplicableToCourse) {
+    return undefined;
+  }
+
+  const hasExpiredSubscriptions = subscriptionsApplicableToCourse.every(
+    subscription => subscription.daysUntilExpirationIncludingRenewals < 0,
+  );
+  const hasExhaustedSubscriptions = subscriptionsApplicableToCourse.every(
+    subscription => subscription?.licenses?.unassigned === 0,
+  );
+  const applicableSubscriptionNonExpiredNonExhausted = subscriptionsApplicableToCourse.find(
+    subscription => subscription.daysUntilExpirationIncludingRenewals >= 0 && subscription?.licenses?.unassigned > 0,
+  );
+
+  if (hasExpiredSubscriptions) {
+    // If customer's subscription plan(s) containing the course being viewed have expired,
+    // change `reasonType` to use the `SUBSCRIPTION_EXPIRED` message.
+    return parseReasonTypeBasedOnEnterpriseAdmins({
+      hasEnterpriseAdminUsers,
+      reasonTypes: {
+        hasAdmins: DISABLED_ENROLL_REASON_TYPES.SUBSCRIPTION_EXPIRED,
+        hasNoAdmins: DISABLED_ENROLL_REASON_TYPES.SUBSCRIPTION_EXPIRED_NO_ADMINS,
+      },
+    });
+  }
+
+  if (hasExhaustedSubscriptions) {
+    // If customer's subscription plan(s) containing the course being viewed no longer have
+    // any remaining seats, change `reasonType` to use the `SUBSCRIPTION_SEATS_EXHAUSTED` message.
+    return parseReasonTypeBasedOnEnterpriseAdmins({
+      hasEnterpriseAdminUsers,
+      reasonTypes: {
+        hasAdmins: DISABLED_ENROLL_REASON_TYPES.SUBSCRIPTION_SEATS_EXHAUSTED,
+        hasNoAdmins: DISABLED_ENROLL_REASON_TYPES.SUBSCRIPTION_SEATS_EXHAUSTED_NO_ADMINS,
+      },
+    });
+  }
+
+  if (subscriptionLicense?.status === LICENSE_STATUS.REVOKED) {
+    // If learner's subscription license is revoked, change `reasonType` to use
+    // the `SUBSCRIPTION_DEACTIVATED` message.
+    return DISABLED_ENROLL_REASON_TYPES.SUBSCRIPTION_DEACTIVATED;
+  }
+
+  if (applicableSubscriptionNonExpiredNonExhausted) {
+    // If customer has a subscription plan(s) containing the course being viewed that is not expired
+    // nor exhausted, change `reasonType` to use the `SUBSCRIPTION_LICENSE_NOT_ASSIGNED` message.
+    return parseReasonTypeBasedOnEnterpriseAdmins({
+      hasEnterpriseAdminUsers,
+      reasonTypes: {
+        hasAdmins: DISABLED_ENROLL_REASON_TYPES.SUBSCRIPTION_LICENSE_NOT_ASSIGNED,
+        hasNoAdmins: DISABLED_ENROLL_REASON_TYPES.SUBSCRIPTION_LICENSE_NOT_ASSIGNED_NO_ADMINS,
+      },
+    });
+  }
+
+  // There is no applicable subscriptions-related reason for disabled enrollment.
+  return undefined;
+};
+
+export const getEnterpriseOffersDisabledEnrollmentReasonType = ({
+  enterpriseOffers,
+  catalogsWithCourse,
+}) => {
+  if (!enterpriseOffers || enterpriseOffers.length === 0) {
+    return DISABLED_ENROLL_REASON_TYPES.NO_ENTERPRISE_OFFERS;
+  }
```

Comment:
Similar to comment below, if there are no offers, I don't believe this function needs to return anything given we wouldn't want to expose a message in the UI saying, "you have no offers" specifically. Similar to how we return `undefined` in `getCouponCodesDisabledEnrollmentReasonType` when there are no coupon codes applicable to the course.

Instead, if this returns `undefined` and there are no other reason types returned for any other subsidies (e.g., subs, etc.), it would fallback to the default reason type of `NO_SUBSIDY` or `NO_SUBSIDY_NO_ADMINS` (depending on whether enterprise customer has administrators configured).